googletest/include/gtest/internal/gtest-port.h

```diff
@@ -604,6 +604,14 @@ struct _RTL_CRITICAL_SECTION;
     || GTEST_OS_QNX || GTEST_OS_FREEBSD || GTEST_OS_NACL)
 #endif  // GTEST_HAS_PTHREAD
 
+// On MinGW, we can have both GTEST_OS_WINDOWS and GTEST_HAS_PTHREAD
+// defined, but we don't want to use MinGW's pthreads implementation, which
+// has conformance problems with some versions of the POSIX standard.
+#if GTEST_HAS_PTHREAD && GTEST_OS_WINDOWS_MINGW
```

Comment:
I don't want to undef something that was possibly set explicitly.
We should just deduce it correctly the first time.

```
#if !defined(GTEST_HAS_PTHREAD) && GTEST_OS_WINDOWS_MINGW
# define GTEST_HAS_PTHREAD 0
#endif
```